qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -0,0 +1,126 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A collections of libraries to setup Calibrations.
+
+Note that the set of available functions will be extended in future releases.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Tuple
+
+from qiskit.circuit import Parameter
+import qiskit.pulse as pulse
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.calibration_management.calibration_key_types import ParameterValueType
+
+
+class BasisGateLibrary(ABC):
+    """A base class for libraries of basis gates to make it easier to setup Calibrations."""
+
+    # Location where default parameter values are stored. These may be updated at construction.
+    __default_values__ = dict()
+
+    def __init__(self, default_values: Optional[Dict] = None):
+        """Setup the library.
+
+        Args:
+            default_values: A dictionary to override library default parameter values.
+        """
+        self._schedules = dict()
+
+        if default_values is not None:
+            for key, value in default_values.items():
+                self.__default_values__[key] = value
+
+    def __getitem__(self, name: str) -> ScheduleBlock:
+        """Return the schedule."""
+        return self._schedules[name]
+
+    @abstractmethod
+    def default_values(self) -> List[Tuple[ParameterValueType, Parameter, Tuple, ScheduleBlock]]:
+        """Return the default values for the parameters.
+
+        Returns
+            A list of tuples is returned. These tuples are structured so that instances of
+            :class:`Calibrations` can call :meth:`add_parameter_value` on the tuples.
+        """
+
+
+class FixedFrequencyTransmonSXRZ(BasisGateLibrary):
```

Comment:
I think just `FixedFrequencyTransmon` if fine. Otherwise we create another library for two qubit? (actually it already have `X` gate and RZ is not defined). Actually it doesn't work because we may call `X` gate from ECR.

Reply:
Good point. See here: 58acc43